apps/snitch_core/test/domain/taxonomy/taxonomy_test.exs

```diff
@@ -0,0 +1,232 @@
+defmodule Snitch.Core.Domain.TaxonomyTest do
+  use ExUnit.Case, async: true
+  use Snitch.DataCase
+
+  import Snitch.Factory
+
+  alias FakerElixir, as: Faker
+  alias Snitch.Domain.Taxonomy
+  alias Snitch.Data.Schema.Taxon
+
+  # Will create a better code for this to create taxonomy from list or map
+  # Creates following taxonomy
+  #   ├── Home & Living
+  #   │   └── Kitchen & Table
+  #   │   └── Table Cover
+  #   ├── Mats & Napkin
+  #   ├── Home decor
+  defp create_taxonomy() do
+    taxonomy = insert(:taxonomy, name: "Home & Living")
+    home_living = insert(:taxon, name: "Home & Living", lft: 0, rgt: 11, taxonomy: taxonomy)
+
+    flooring =
+      insert(
+        :taxon,
+        name: "Flooring",
+        lft: 1,
+        rgt: 2,
+        parent_id: home_living.id,
+        taxonomy: taxonomy
+      )
+
+    kitchen_table =
+      insert(
+        :taxon,
+        name: "Kitchen & Tables",
+        lft: 3,
+        rgt: 8,
+        parent_id: home_living.id,
+        taxonomy: taxonomy
+      )
+
+    home_decor =
+      insert(
+        :taxon,
+        name: "Home Decor",
+        lft: 9,
+        rgt: 10,
+        parent_id: home_living.id,
+        taxonomy: taxonomy
+      )
+
+    table_cover =
+      insert(
+        :taxon,
+        name: "Table Covers",
+        lft: 4,
+        rgt: 5,
+        parent_id: kitchen_table.id,
+        taxonomy: taxonomy
+      )
+
+    mat_napkins =
+      insert(
+        :taxon,
+        name: "Mat & Napkins",
+        lft: 6,
+        rgt: 7,
+        parent_id: kitchen_table.id,
+        taxonomy: taxonomy
+      )
+
+    {home_living,
+     [
+       {flooring, []},
+       {kitchen_table,
+        [
+          {table_cover, []},
+          {mat_napkins, []}
+        ]},
+       {home_decor, []}
+     ]}
+  end
+
+  describe "add_taxon/2" do
+    test "add child taxon" do
+      {_, [{target, _} | _]} = create_taxonomy()
+
+      child = %Taxon{name: "Carpet"}
+      taxon = Taxonomy.add_taxon(target, child, :child)
+
+      taxonomies = get_taxons_name_list(target)
+
+      assert taxon.name == "Carpet"
+      assert taxon.taxonomy_id == target.taxonomy_id
+      assert taxon.parent_id == target.id
+
+      assert taxonomies == [
```

Comment:
Technically, an inorder traversal does not uniquely determine the tree structure (two different traversals are needed) It's best to match on the n-ary tuple (of just names, perhaps).